### PR TITLE
CLM -  Add built time column to the content environment target and show it on the UI

### PR DESCRIPTION
**Pull request by lneves12**
## What does this PR change?

CLM -  Add built time column to the content environment target and show it on the UI

## GUI diff

Screenshot from 2019-05-17 17-18-22

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- No tests

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/7808

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests" 		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

**Comment by lneves12** [comment]
This is ready for review! :)

File: schema/spacewalk/upgrade/susemanager-schema-4.0.11-to-susemanager-schema-4.0.12/051-suseContentEnvironmentTarget.sql.oracle [hunk 1]
```diff
@@ -0,0 +1 @@
+-- intentionally blank, corresponds to 20-environment-target-state.sql.postgresql
```
Comment (ncounter) [inline schema/spacewalk/upgrade/susemanager-schema-4.0.11-to-susemanager-schema-4.0.12/051-suseContentEnvironmentTarget.sql.oracle]: Maybe this was a copy/paste leftover and it was intended to be `051-suseContentEnvironmentTarget.sql.postgresql`?

**Comment by lneves12** [comment]
@ncounter done, also improved the method computeBuiltTime to return a Date only if the environment state is BUILT

File: java/code/src/com/suse/utils/Opt.java [hunk 2]
```diff
@@ -122,4 +123,13 @@ private Opt() {
             return second.get();
         }
     }
+
+    /**
+     * Return an empty optional date
+     * @return optional empty date
+     */
+    public static Optional<Date> emptyDate() {
```
[the remark this answers is not in the document]
Comment (lneves12) [reply java/code/src/com/suse/utils/Opt.java]: For some reason with my solution calling directly Optional.empty() wasn't working. Java couldn't infer the optional type and it wasn't even compiling

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentEnvironment.java [hunk 3]
```diff
@@ -321,21 +323,46 @@ public String toString() {
...
+    /**
+     * Computes the built time of the {@link ContentEnvironment} based on the built times of its
+     * {@link EnvironmentTarget}
+     * Most recent built time of all the environment targets
+     *
+     * @return the built time
+     */
+    public Optional<Date> computeBuiltTime() {
+        return this.computeStatus()
+                .map(envStatus -> {
```
[the remark this answers is not in the document]
Comment (lneves12) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/ContentEnvironment.java]: Thanks for the tip :+1:

**Comment by lneves12** [comment]
@hustodemon thanks for the review and tips! I have addressed all your comments

**Pull request merged**